_pytest/mark.py

```diff
@@ -4,17 +4,61 @@
 from operator import attrgetter
 from .compat import imap
 
+
 def alias(name):
     return property(attrgetter(name), doc='alias for ' + name)
 
 
+class ParameterSet(namedtuple('ParameterSet', 'values, marks, id')):
+    @classmethod
+    def param(cls, *value, **kw):
+        marks = kw.pop('marks', ())
+        
+        if isinstance(marks, MarkDecorator):
+            marks = marks,
+        else:
+            assert isinstance(marks, (tuple, list, set))
```

Comment:
Oh OK, I agree. 

Now I see, I got confused by your comment on the line below:

> while taking a look again, it may be sensible to trigger a pytest warning there

and thought you meant a warning here as well. My bad.

Having collection error at this point is definitely the proper approach. 👍